### ISSUE TITLE
redundantSelf rule inserts self before the parameter name

### ISSUE THREAD

**Issue by ojii** [nicklockwood/SwiftFormat#417]
With 0.40.6 `--self insert` sometimes inserts `self` in front of the name of parameters in function calls from within classes which have a property of the same name.

This is the example of a diff it produced:

```diff
-        switch Policy.local.checkRequest(request: navigationAction.request) {
+        switch Policy.local.checkRequest(self.request: navigationAction.request) {
```

That's from inside a method in a class with a `request` property.

**Comment by nicklockwood** [comment]
Thanks for reporting, and for the PR! I'll fix the test issues and make a new release.

**Comment by nicklockwood** [comment]
@ojii this is now fixed on the [develop](https://github.com/nicklockwood/SwiftFormat/tree/develop) branch.

**Comment by nicklockwood** [comment]
@ojii fixed in 0.40.7
